test/doctool/test-apilinks.js

```diff
@@ -2,23 +2,36 @@
 
 require('../common');
 const fixtures = require('../common/fixtures');
+const fs = require('fs');
 const assert = require('assert');
 const path = require('path');
 const { execFileSync } = require('child_process');
 
 const script = path.join(__dirname, '..', '..', 'tools', 'doc', 'apilinks.js');
-const fixture = fixtures.path('apilinks', 'known1.js');
-const expected = {
-  'Known.classField': 'known1.js#L7',
-  'known1.createKnown': 'known1.js#L9'
-};
-const ret = execFileSync(
-  process.execPath, [script, fixture], { encoding: 'utf-8' }
-);
-const links = JSON.parse(ret);
-for (const [k, v] of Object.entries(expected)) {
-  assert.ok(k in links);
-  assert.ok(links[k].includes(v));
-  delete links[k];
-}
-assert.strictEqual(Object.keys(links).length, 0);
+
+const apilinks = fixtures.path('apilinks');
+fs.readdirSync(apilinks).forEach((fixture) => {
+  if (!fixture.endsWith('.js')) return;
+  const file = path.join(apilinks, fixture);
+
+  let expected = {};
+  eval(/(expected\s*=.*?);/s.exec(fs.readFileSync(file, 'utf8'))[1]);
```

Comment:
A bit of a dirty trick...
1. We consider any use of `eval` as lint
2. We want test code to be K.I.S.S. so we don't what to test out tests

So in this case I would rather have a hardcoded:
```js
const expectedInFixtures = {
  'known1.js': {
    'Known.classField': 'known1.js#L7',
    'known1.createKnown': 'known1.js#L9',
  },
};
```

Or if you really want a generic tester wyou could delineate the expected output in the fixtures, slice it out, and `JSON.parse` it.
assuming fixtures end with:
```js
/* START EXPECTED
{
  "Known.classField": "known1.js#L7",
  "known1.createKnown": "known1.js#L9"
}
  END EXPECTED */
```
run code could be
```js
const fixtureContext = fs.readFileSync(file, 'utf8');
const fixtureExpected = /\/\* START EXPECTED(.*)  END EXPECTED \*\//s.exec(fixtureContext);
const expectedlinks = JSON.parse(fixtureExpected[1]);
```